android/src/app/organicmaps/location/LocationHelper.java

```diff
@@ -481,40 +319,30 @@ public void restart()
   {
     Logger.d(TAG);
     stop();
+    if (ContextCompat.checkSelfPermission(mContext, ACCESS_COARSE_LOCATION) != PERMISSION_GRANTED &&
```

Comment:
1. Are both really required here? Or coarse could be enough?
2. What if users deliberately deny permissions in system settings and want to use OM without location?